client/src/pages/forms/FormBase.tsx

```diff
@@ -0,0 +1,152 @@
+            Details
+          </Typography>
+          <Grid container spacing={2}>
+            <Grid item xs={12}>
+              <Controller
+                name="title"
+                control={control}
+                rules={{
+                  required: true,
+                  minLength: 1,
+                }}
+                render={({
+                  field: { onChange, value },
+                  fieldState: { invalid, error },
+                }) => (
+                  <TextField
+                    fullWidth
+                    label="Title"
+                    variant="outlined"
+                    onChange={onChange}
+                    value={value}
+                    required
+                    error={invalid}
+                    helperText={getFormErrorMessage(error?.type)}
+                  />
+                )}
+              />
+            </Grid>
+            <Grid item xs={12}>
+              <Controller
+                name="description"
+                control={control}
+                rules={{
+                  required: true,
+                  minLength: 1,
+                }}
+                render={({
+                  field: { onChange, value },
+                  fieldState: { invalid, error },
+                }) => (
+                  <TextField
+                    fullWidth
+                    label="Description"
+                    variant="outlined"
+                    onChange={onChange}
+                    value={value}
+                    required
+                    error={invalid}
+                    helperText={getFormErrorMessage(error?.type)}
+                  />
+                )}
+              />
+            </Grid>
+            <Grid item xs={12}>
+              <Controller
+                name="purpose"
+                control={control}
+                rules={{
+                  required: true,
+                }}
+                render={({
+                  field: { onChange, value },
+                  fieldState: { invalid, error },
+                }) => (
+                  <FormControl fullWidth required error={!!error}>
+                    <InputLabel id="demo-simple-select-label">
+                      Purpose
+                    </InputLabel>
+                    <Select
+                      id="purpose"
+                      label="Purpose"
+                      onChange={onChange}
+                      value={value}
+                      error={invalid}
+                    >
+                      <MenuItem value={FormPurpose.ACTIVITY_ENTRY}>
+                        Activity Entry
+                      </MenuItem>
+                      <MenuItem value={FormPurpose.CUSTOMER}>Customer</MenuItem>
+                      <MenuItem value={FormPurpose.INVENTORY}>
+                        Inventory
+                      </MenuItem>
+                      <MenuItem value={FormPurpose.STAFF}>Staff</MenuItem>
```

Comment:
These would need more context, added as a separate ticket for also integrating the form builder